test/envoye2e/driver/fake_stackdriver.go

```diff
@@ -191,9 +203,98 @@ func (s *LoggingServer) GetLogEntries(w http.ResponseWriter, req *http.Request)
 	}
 }
 
+// ListTraces implements ListTraces method.
+func (s *TracesServer) ListTraces(context.Context, *cloudtracev1.ListTracesRequest) (*cloudtracev1.ListTracesResponse, error) {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+	return &s.listTracesResp, nil
+}
+
+// GetTrace implements GetTrace method.
+func (s *TracesServer) GetTrace(context.Context, *cloudtracev1.GetTraceRequest) (*cloudtracev1.Trace, error) {
+	log.Fatal("Unimplemented Method")
+	return &cloudtracev1.Trace{}, nil
+}
+
+// GetTrace implements GetTrace method.
+func (s *TracesServer) PatchTraces(context.Context, *cloudtracev1.PatchTracesRequest) (*empty.Empty, error) {
+	log.Fatal("Unimplemented Method")
+	return &empty.Empty{}, nil
+}
+
+func getID(id string) (uint64, error) {
+	// Convert hexadecimal string to int64
+	dec, err := strconv.ParseInt(id, 16, 64)
+	if err != nil {
+		return 0, err
+	}
+	return uint64(dec), nil
+}
+
+// BatchWriteSpans implements BatchWriteSpans method.
+func (s *TracesServer) BatchWriteSpans(ctx context.Context, req *cloudtracev2.BatchWriteSpansRequest) (*empty.Empty, error) {
+	log.Printf("receive BatchWriteSpansRequest %+v", *req)
+	s.mux.Lock()
+	defer s.mux.Unlock()
+	// Map that stores individual traces.
+	m := make(map[string]*cloudtracev1.Trace)
```

Comment:
This does not seem to be right. Spans from one trace won't be included in the same batchWriteSpansRequest. I think what you probably should do is keep the map in TracesServer, add new span to it on every batch write reqeust, and create a `listTracesResp` with the map at the end of this method or in the ListTraces method. This way response would have all the spans received correctly organized (spans from the same trace end up in the same trace object in ListTraceResp).

Reply:
Thanks.. so if spans from one trace can come in multiple requests.. it can happen that List call is called before all the spans are received...
In that case, before sending a trace in listtrace we should make sure that atleast parent trace is received?